provides.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from charms import reactive
+
+
+class BindClientProvides(reactive.Endpoint):
```

Comment:
I wonder if this interface should implement a method to set attributes (passed via arg) to be shared over the relation. For example, the [`http` interface](https://github.com/juju-solutions/interface-http/blob/master/provides.py#L25) implements a `configure` method.

Reply:
I think it's a nice idea and I implemented to pass the port with a `configure` method. I've also added in the README how to use it.